### ISSUE TITLE
suggestion: view=unified = default +hotkey request

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#239]
the label -switching mode is quite superior already IMO.. i think it would be a better default. I guess you can still get at the simpler mode through the search (someone focussed on a specific task would use that)

using my desktop machine for a bit,

keys [ , ] = next/previous label (or some other suitable.. i remember those from the paint-program DPaint) to step through the label palette. that would give you a quickway to ch check everything

Once you can add labels in this view swell..  the tool will be at a new level of useability

**Comment by bbernhard** [comment]
> the label -switching mode is quite superior already IMO.. i think it would be a better default. I guess you can still get at the simpler mode through the search (someone focussed on a specific task would use that)

yeah, the current Annotation dropdown menu is not particular nice :/. The idea was to add different UI views for different target groups: 

* power user: probably (browse-based) unified mode
* mobile user: probably default mode
* tablet user: probably (browse-based) default mode

But supporting all those different combinations makes the dropdown look crowded. :/ Of course, we could also do some sort of autodetection and try to automatically choose the right mode depending on the device. But not sure how reliable that works. 

If possible, I would like to keep the mode selection somewhere, as it is also a reminder for me to regression-test the different modes before a new release. If the modes are deeply hidden inside a menu, I'll probably forget about those and at some point they are completely broken. :/ (unfortunately I have to test most of the UI functionality by hand, as Selenium creates too many false-positives due to timing issues in the CI environment)

But I totally agree with you, the current solution is sub-optimal. I am open for any suggestions ;)
Maybe we can change the behavior, so that the dropdown menu only opens when one clicks on the arrow and if someone clicks on the text we open the default annotation mode (=unified mode)? 
> keys [ , ] = next/previous label (or some other suitable.. i remember those from the paint-program DPaint) to step through the label palette. that would give you a quickway to ch check everything

just out of interest, are you left handed? 

Initially I thought about using the <tab> key for switching between the different labels. But that of course works better for right handed people...left handed people would probably prefer a key that's positioned on the right half of the keyboard. (like `,`)

**Comment by dobkeratops** [comment]
stopgap suggestion: Would it be possible to add mutual shortcuts between the Add Labels page and Unified view.. i.e. buttons for "open this image in add labels" <-> "open this image in unified annotation". That would let you fill the gaps and get back to really annotate them


> just out of interest, are you left handed?

right handed; dominant hand on the mouse, left hand on the keyboard.
[,] are used in modern paint programs to modify brush size or some other focussed slicder (eg opacity). An alternate like < , > might be ok. Without brush-size, [.] is available for palette flicking.

X is often used for FG/BG color toggle (swap last 2 labels?)

alt-click is often used to pick a colour from the image (copy the label under this?)

**Comment by dobkeratops** [comment]
Few more hotkey requests:-

(i) in Add Labels mode: a hotkey for Done - 'shift+Enter' is 'add', perhaps 'ctrl+enter' could be Done
(ii) add the same 'add label' hotkey to Unified Mode, so that adding many labels together is just as fast
(iib) if you could display free labels in Unified Mode, you could use this *instead* of the dedicated add labels screen.. then you can just decide what to do to each image.. if there are no labels to add, just get on with annotating, otherwise add the labels.. 
(iii) a hotkey to focus the label addition text entry box - perhaps just pressing Enter without a modifier? (thats what i did in my annotation tool experiment.. there was just a 'current label' indicator, and you just pressed enter to bring up a text entry modal dialogue box to change the current label)
